### ISSUE TITLE
"class template argument deduction failed" compiling NCF4.cpp in Ubuntu 18.04

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#2417]
I'm getting the following compiling the `development` branch in Ubuntu 18.04 (default GCC version is 7.5.0):

```
g++ -pipe -Wall -Wno-shadow -Wcast-qual -Wno-sign-conversion -Wno-sign-compare -Wno-parentheses -Wno-sign-compare  -Wuninitialized -I/usr/include  -Wno-error=pragmas -std=gnu++14 -g3 -O2  -Wno-unused-local-typedefs -Wno-unknown-pragmas -std=gnu++17 -Wno-mismatched-tags -I../../../../Macaulay2/e -DNDEBUG -I/root/M2/M2/BUILD/doug/../../include -I/root/M2/M2/BUILD/doug/include -I/root/M2/M2/BUILD/doug/usr-host/include -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest   -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -I../d -I../../../../Macaulay2/e/../c -I../../../../Macaulay2/e/../d -Wno-fatal-errors -Wno-unused-variable  -c -Wno-cast-qual ../../../../Macaulay2/e/NCAlgebras/NCF4.cpp -o NCAlgebras/NCF4.o
In file included from /usr/include/mathicgb/mtbb.hpp:21:0,
                 from ../../../../Macaulay2/e/m2tbb.hpp:7,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.hpp:4,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.cpp:1:
/usr/include/tbb/tbb.h:21:154: note: #pragma message: TBB Warning: tbb.h contains deprecated functionality. For details, please see Deprecated Features appendix in the TBB reference manual.
 details, please see Deprecated Features appendix in the TBB reference manual.")
                                                                               ^
In file included from ../../../../Macaulay2/e/aring-translate.hpp:18:0,
                 from ../../../../Macaulay2/e/aring-glue.hpp:7,
                 from ../../../../Macaulay2/e/VectorArithmetic.hpp:9,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.hpp:14,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.cpp:1:
../../../../Macaulay2/e/aring-zzp-flint.hpp:19:32: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wmacro-redefined"
                                ^~~~~~~~~~~~~~~~~~~
In file included from ../../../../Macaulay2/e/aring-translate.hpp:19:0,
                 from ../../../../Macaulay2/e/aring-glue.hpp:7,
                 from ../../../../Macaulay2/e/VectorArithmetic.hpp:9,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.hpp:14,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.cpp:1:
../../../../Macaulay2/e/aring-zzp-ffpack.hpp:15:32: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wmacro-redefined"
                                ^~~~~~~~~~~~~~~~~~~
In file included from ../../../../Macaulay2/e/aring-translate.hpp:23:0,
                 from ../../../../Macaulay2/e/aring-glue.hpp:7,
                 from ../../../../Macaulay2/e/VectorArithmetic.hpp:9,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.hpp:14,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.cpp:1:
../../../../Macaulay2/e/aring-gf-flint-big.hpp:13:32: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wmacro-redefined"
                                ^~~~~~~~~~~~~~~~~~~
In file included from ../../../../Macaulay2/e/NCAlgebras/NCF4.hpp:14:0,
                 from ../../../../Macaulay2/e/NCAlgebras/NCF4.cpp:1:
../../../../Macaulay2/e/VectorArithmetic.hpp: In constructor 'VectorArithmetic::VectorArithmetic(const Ring*)':
../../../../Macaulay2/e/VectorArithmetic.hpp:413:85: error: class template argument deduction failed:
    (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZpFlint>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:413:85: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:413:85: note:   candidate expects 2 arguments, 0 provided
    (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZpFlint>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:413:85: note:   couldn't deduce template parameter 'RingType'
    (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZpFlint>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:417:87: error: class template argument deduction failed:
  (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFFlintBig>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:417:87: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:417:87: note:   candidate expects 2 arguments, 0 provided
  (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFFlintBig>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:417:87: note:   couldn't deduce template parameter 'RingType'
  (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFFlintBig>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:421:84: error: class template argument deduction failed:
     (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFFlint>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:421:84: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:421:84: note:   candidate expects 2 arguments, 0 provided
     (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFFlint>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:421:84: note:   couldn't deduce template parameter 'RingType'
     (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFFlint>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:425:82: error: class template argument deduction failed:
       (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingQQGMP>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:425:82: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:425:82: note:   candidate expects 2 arguments, 0 provided
       (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingQQGMP>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:425:82: note:   couldn't deduce template parameter 'RingType'
       (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingQQGMP>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:429:86: error: class template argument deduction failed:
   (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZpFFPACK>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:429:86: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:429:86: note:   candidate expects 2 arguments, 0 provided
   (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZpFFPACK>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:429:86: note:   couldn't deduce template parameter 'RingType'
   (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZpFFPACK>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:433:80: error: class template argument deduction failed:
         (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZp>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:433:80: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:433:80: note:   candidate expects 2 arguments, 0 provided
         (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZp>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:433:80: note:   couldn't deduce template parameter 'RingType'
         (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingZZp>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:437:81: error: class template argument deduction failed:
        (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFM2>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:437:81: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:437:81: note:   candidate expects 2 arguments, 0 provided
        (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFM2>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:437:81: note:   couldn't deduce template parameter 'RingType'
        (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFM2>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:441:85: error: class template argument deduction failed:
    (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFGivaro>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:441:85: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:441:85: note:   candidate expects 2 arguments, 0 provided
    (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFGivaro>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:441:85: note:   couldn't deduce template parameter 'RingType'
    (R, &dynamic_cast< const M2::ConcreteRing<M2::ARingGFGivaro>* >(R)->ring());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:445:96: error: class template argument deduction failed:
 eVector = new ConcreteVectorArithmetic(R, R->cast_to_Z_mod()->get_CoeffRing());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:445:96: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:445:96: note:   candidate expects 2 arguments, 0 provided
 eVector = new ConcreteVectorArithmetic(R, R->cast_to_Z_mod()->get_CoeffRing());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:445:96: note:   couldn't deduce template parameter 'RingType'
 eVector = new ConcreteVectorArithmetic(R, R->cast_to_Z_mod()->get_CoeffRing());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:449:87: error: class template argument deduction failed:
    mConcreteVector = new ConcreteVectorArithmetic(R, R->getCoefficientRingR());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:449:87: error: no matching function for call to 'ConcreteVectorArithmetic()'
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic(const Ring*, const RingType*)-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic(const Ring* origR, const RingType* R) : mOriginalRing(origR), mRing(R) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:68:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:449:87: note:   candidate expects 2 arguments, 0 provided
    mConcreteVector = new ConcreteVectorArithmetic(R, R->getCoefficientRingR());
                                                                              ^
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note: candidate: template<class RingType, class ElementArrayType> ConcreteVectorArithmetic()-> ConcreteVectorArithmetic<RingType, ElementArrayType>
   ConcreteVectorArithmetic() : mOriginalRing(nullptr), mRing(nullptr) {}
   ^~~~~~~~~~~~~~~~~~~~~~~~
../../../../Macaulay2/e/VectorArithmetic.hpp:56:3: note:   template argument deduction/substitution failed:
../../../../Macaulay2/e/VectorArithmetic.hpp:449:87: note:   couldn't deduce template parameter 'RingType'
    mConcreteVector = new ConcreteVectorArithmetic(R, R->getCoefficientRingR());
                                                                              ^
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-mismatched-tags'
Makefile.common:32: recipe for target 'NCAlgebras/NCF4.o' failed
make[2]: *** [NCAlgebras/NCF4.o] Error 1
make[2]: Leaving directory '/root/M2/M2/BUILD/doug/Macaulay2/e'
Makefile:15: recipe for target 'all-in-e' failed
make[1]: *** [all-in-e] Error 2
make[1]: Leaving directory '/root/M2/M2/BUILD/doug/Macaulay2'
GNUmakefile:228: recipe for target 'all-in-Macaulay2' failed
make: *** [all-in-Macaulay2] Error 2
```

The above is from a local build, but this is also happening on PPA builds, e.g., [this one](https://launchpadlibrarian.net/590894831/buildlog_ubuntu-bionic-amd64.macaulay2_1.19.1.1+git202203151450-0ppa202203151518~ubuntu18.04.1_BUILDING.txt.gz).

**Comment by mikestillman** [comment]
These are c++17 things.  I wonder why there are two -std= lines, and they are different...  But usually the later one takes precedence, I think.  gcc 7 is pretty old now, I think...  I'm surprised it OK's -std=gnu++17.

Is it possible to use a more recent compiler?  Frank and I might remove this construct, but we were not sure if we wanted to do that.

**Comment by d-torrance** [comment]
> These are c++17 things. I wonder why there are two -std= lines, and they are different...

`-std=gnu++14` is set by `configure` for all of the directories, but `-std=gnu++17` is set in the `e` directory `Makefile`:
https://github.com/Macaulay2/M2/blob/0b0159476fbb752b5c0bfd9a50a50b41c0cf95a5/M2/configure.ac#L251-L253
https://github.com/Macaulay2/M2/blob/0b0159476fbb752b5c0bfd9a50a50b41c0cf95a5/M2/Macaulay2/e/Makefile.in#L13

Maybe it would make sense to just set `-std=gnu++17` in `configure` and be done with it?

> But usually the later one takes precedence, I think. gcc 7 is pretty old now, I think... I'm surprised it OK's -std=gnu++17.

According to https://gcc.gnu.org/projects/cxx-status.html#cxx17, GCC 7 supported most of C++17.

> Is it possible to use a more recent compiler?

GCC 8 is also available in Ubuntu 18.04, but I'm getting the same error compiling with it.  :\
